### PR TITLE
Fix error showing when opening a job file for extenders

### PR DESCRIPTION
**Pull request by rudyflores**
Signed-off-by: Rudy Flores <68666202+rudyflores@users.noreply.github.com>

## Proposed changes

Fix an issue where a job file is open and an error is shown mentioning the text document was not able to be shown even though it is open successfully for extenders.

## Release Notes

Milestone:

Changelog: Fix error displayed when opening a job file for extenders

## Types of changes

What types of changes does your code introduce to Zowe Explorer?
_Put an `x` in the boxes that apply_

- [x] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Updates to Documentation or Tests (if none of the other choices apply)

## Checklist

_Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This checklist will be used as reference for both the contributor and the reviewer_

- [x] I have read the [CONTRIBUTOR GUIDANCE](https://github.com/zowe/vscode-extension-for-zowe/wiki/Best-Practices:-Contributor-Guidance) wiki
- [x] PR title follows [Conventional Commits Guidelines](https://www.conventionalcommits.org/en/v1.0.0-beta.2/)
- [x] PR Description is included
- [ ] gif or screenshot is included if visual changes are made
- [x] `yarn workspace vscode-extension-for-zowe vscode:prepublish` has been executed
- [x] All checks have passed (DCO, Jenkins and Code Coverage)
- [ ] I have added unit test and it is passing
- [ ] I have added integration test and it is passing
- [ ] There is coverage for the code that I have added
- [x] I have tested it manually and there are no regressions found
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any PR dependencies have been merged and published (if appropriate)

## Reminder

After a PR is merged into the `master` branch, create a PR from `master` to the `next` branch resolving any conflicts.

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Review by JillieBeanSim — DISMISSED** [review]
I was able to recreate the issue in CRW with the marketplace version and tested with this branch and the issue is fixed.  Thanks @rudyflores for your work on this

**Comment by JillieBeanSim** [comment]
merged in next and rebased the PR

File: packages/zowe-explorer/src/job/actions.ts [hunk 1]
```diff
@@ -73,10 +73,16 @@ export async function getSpoolContent(session: string, spool: zowe.IJobFile, ref
...
     if (profiles.validProfile === ValidProfileEnum.VALID || profiles.validProfile === ValidProfileEnum.UNVERIFIED) {
-        const uri = toUniqueJobFileUri(session, spool)(refreshTimestamp.toString());
+        const uri: vscode.Uri = toUniqueJobFileUri(session, spool)(refreshTimestamp.toString());
         try {
             await vscode.window.showTextDocument(uri);
         } catch (error) {
+            const isTextDocActive =
+                vscode.window.activeTextEditor.document.uri?.path === `${spool.jobname}.${spool.jobid}.${spool.ddname}`;
+
+            if (String(error.message).includes("Failed to show text document") && isTextDocActive) {
```
Comment (phaumer) [inline packages/zowe-explorer/src/job/actions.ts:83]: Will this work if the user works in a different language (as Theia is now localized)?

**Review by phaumer — APPROVED** [review]
This is a fix that will quickly remove the error message, but the solution is not very robust as it just looks for an English substring of the error message. 

We can merge it for a quick fix, but a the same time we should fix the root cause of the problem to avoid these errors all together. 

One suspicion I have is that the format of the URI might work with VS Code, but causes issues in Theia/Che because it is not a regular URI. When I serialize one in the debugger it looks like this:

```
zosspool:DBB.STC00021.JESMSGLG?["zosdevtestexp22.rseapi",{"id":2,"ddname":"JESMSGLG","procstep":"","record-count":"2","stepname":"JES2","jobid":"STC00021","jobname":"DBB"}]#1651622167209
```

Perhaps the solution is to revise the uri schema used for spool files?

File: packages/zowe-explorer/src/job/actions.ts (hunk 1 above)
Comment (rudyflores) [reply packages/zowe-explorer/src/job/actions.ts:83]: You have a good point here, might have to look into how I could fetch that error localized and check for it...

Comment (kristinochka) [reply packages/zowe-explorer/src/job/actions.ts:83]: Check if vscode api defines that error as an error code and then check against that code. The message string will likely change in the future.

**Review by phaumer — APPROVED** [review]

**Pull request merged**